app/upgrades/v15/upgrades.go

```diff
@@ -59,6 +66,48 @@ func setICQParams(ctx sdk.Context, icqKeeper *icqkeeper.Keeper) {
 	icqKeeper.SetParams(ctx, icqparams)
 }
 
+func migrateBalancerPoolsToSolidlyStable(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, poolmanagerKeeper *poolmanager.Keeper) {
+	// migrate stOSMO_OSMOPoolId, stJUNO_JUNOPoolId, stSTARS_STARSPoolId
+	pools := []uint64{stOSMO_OSMOPoolId, stJUNO_JUNOPoolId, stSTARS_STARSPoolId}
+	for _, poolId := range pools {
+		migrateBalancerPoolToSolidlyStable(ctx, gammKeeper, poolmanagerKeeper, poolId)
+	}
+}
+
+func migrateBalancerPoolToSolidlyStable(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, poolmanagerKeeper *poolmanager.Keeper, poolId uint64) {
+	// fetch the pool with the given poolId
+	balancerPool, err := gammKeeper.GetPool(ctx, poolId)
+	if err != nil {
+		panic(err)
+	}
+
+	// initialize the stableswap pool
+	stableswapPool, err := stableswap.NewStableswapPool(
+		poolId,
+		stableswap.PoolParams{SwapFee: balancerPool.GetSwapFee(ctx), ExitFee: balancerPool.GetExitFee(ctx)},
+		balancerPool.GetTotalPoolLiquidity(ctx),
+		[]uint64{1, 1},
+		"osmo1k8c2m5cn322akk5wy8lpt87dd2f4yh9afcd7af", // Stride Foundation 2/3 multisig
+		"",
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// ensure the number of stableswap LP shares is the same as the number of balancer LP shares
+	totalShares := sdk.NewCoin(
+		gammtypes.GetPoolShareDenom(poolId),
+		balancerPool.GetTotalShares(),
+	)
+	stableswapPool.TotalShares = totalShares
```

Comment:
We should also think about whether TWAP records need to be handled. No action is needed at the moment, just leaving this as a reminder to confirm before the merge

Reply:
I personally don't think they need to be handled, do you think we might have the possibility that we need to handle this?

Reply:
Agreed, I think that's fine